python/mmSolver/utils/reproject.py

```diff
@@ -82,22 +82,35 @@ def create_reprojection_on_camera(cam_tfm, cam_shp):
     maya.cmds.connectAttr(cam_tfm + '.worldMatrix', node + '.cameraWorldMatrix')
     maya.cmds.connectAttr(cam_shp + '.focalLength', node + '.focalLength')
     maya.cmds.connectAttr(cam_shp + '.cameraAperture', node + '.cameraAperture')
-    maya.cmds.connectAttr(cam_shp + '.filmOffset', node + '.filmOffset')
     maya.cmds.connectAttr(cam_shp + '.filmFit', node + '.filmFit')
     maya.cmds.connectAttr(cam_shp + '.nearClipPlane', node + '.nearClipPlane')
     maya.cmds.connectAttr(cam_shp + '.farClipPlane', node + '.farClipPlane')
     maya.cmds.connectAttr(cam_shp + '.cameraScale', node + '.cameraScale')
 
     # Connect render settings attributes
     resolution_factor = 10000.0
-    mult_node = maya.cmds.createNode('multiplyDivide')
+    mult_node = maya.cmds.createNode('multiplyDivide', name='main_multiplyDivide1')
     maya.cmds.setAttr(mult_node + '.input2X', resolution_factor)
     maya.cmds.setAttr(mult_node + '.input2Y', resolution_factor)
     maya.cmds.connectAttr(cam_shp + '.horizontalFilmAperture', mult_node + '.input1X')
     maya.cmds.connectAttr(cam_shp + '.verticalFilmAperture', mult_node + '.input1Y')
     maya.cmds.connectAttr(mult_node + '.outputX', node + '.imageWidth')
     maya.cmds.connectAttr(mult_node + '.outputY', node + '.imageHeight')
 
+    # create 2d offset setup
+    offset_plus_minus_node = maya.cmds.createNode('plusMinusAverage', name='offset_plusMinusAverage1')
+    maya.cmds.connectAttr(cam_shp + '.filmOffset', offset_plus_minus_node + '.input2D[0]')
+    maya.cmds.setAttr(offset_plus_minus_node + '.input2D[1]', 0.0, 0.0, type='float2')
+    maya.cmds.connectAttr(offset_plus_minus_node + '.output2D', node + '.filmOffset')
+
+    # create a zoom setup
+    zoom_mult_node = maya.cmds.createNode('multiplyDivide', name='zoom_multiplyDivide1')
+    maya.cmds.setAttr(zoom_mult_node + '.input1X', 1.0)
+    expression_string = '{zoom_mult_node}.input2X = 1/{zoom_mult_node}.input1X;'.format(zoom_mult_node=zoom_mult_node)
+    maya.cmds.expression(string=expression_string, object=zoom_mult_node)
```

Comment:
I know it's tempting to use an expression and I know it will work, however I'd prefer to avoid expressions because they will slow down the Maya scene evaluation and stop parallel evaluation. Another reason is that expressions are only evaluated when the user changes frame, but node connections will evaluate immediately.

It should be possible to replace the `1/{zoom_mult_node}.input1X` with yet another "multiplyDivide" node with a "divide" operation to provide the same output value.